### ISSUE TITLE
Ingress addon problem

### ISSUE THREAD

**Issue by AndreyIvshin** [kubernetes/minikube#10152]
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
1. Starting minikube == 4 step of https://github.com/kubernetes/minikube/issues/10151 Docker driver minikube cannot be created after enabling ingress addon #10151
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

$ minikube start
* minikube v1.16.0 on Microsoft Windows 10 Home Single Language 10.0.19041 Build 19041
* Automatically selected the docker driver
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Creating docker container (CPUs=2, Memory=1993MB) ...
! Listening to 0.0.0.0 on external docker host 192.168.99.100. Please be advised
! StartHost failed, but will try again: creating host: create: creating: setting up container node: preparing volume for minikube container: docker run --rm --entrypoint /usr/bin/test -v minikube:/var gcr.io/k8s-minikube/kicbase:v0.
0.15-snapshot4@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16 -d /var/lib: exit status 125
stdout:

stderr:
Unable to find image 'gcr.io/k8s-minikube/kicbase:v0.0.15-snapshot4@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16' locally
sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16: Pulling from k8s-minikube/kicbase
Digest: sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16
Status: Downloaded newer image for gcr.io/k8s-minikube/kicbase@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16
docker: Error response from daemon: cgroups: cannot find cgroup mount destination: unknown.
time="2021-01-15T21:39:37+06:00" level=error msg="error waiting for container: context canceled"

* docker "minikube" container is missing, will recreate.
* Creating docker container (CPUs=2, Memory=1993MB) ...
! Listening to 0.0.0.0 on external docker host 192.168.99.100. Please be advised
* Failed to start docker container. Running "minikube delete" may fix it: recreate: creating host: create: creating: setting up container node: preparing volume for minikube container: docker run --rm --entrypoint /usr/bin/test -v m
inikube:/var gcr.io/k8s-minikube/kicbase:v0.0.15-snapshot4@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16 -d /var/lib: exit status 125
stdout:

stderr:
docker: Error response from daemon: cgroups: cannot find cgroup mount destination: unknown.

! Startup with docker driver failed, trying with alternate driver virtualbox: Failed to start host: recreate: creating host: create: creating: setting up container node: preparing volume for minikube container: docker run --rm --ent
rypoint /usr/bin/test -v minikube:/var gcr.io/k8s-minikube/kicbase:v0.0.15-snapshot4@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb31649c16 -d /var/lib: exit status 125
stdout:

stderr:
docker: Error response from daemon: cgroups: cannot find cgroup mount destination: unknown.

* Deleting "minikube" in docker ...
* Removing C:\Users\graog\.minikube\machines\minikube ...
! Failed to delete cluster minikube, proceeding with retry anyway.
* Starting control plane node minikube in cluster minikube
* Creating virtualbox VM (CPUs=2, Memory=2200MB, Disk=20000MB) ...
* Found network options:
  - NO_PROXY=192.168.99.100
  - no_proxy=192.168.99.100
* Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
  - env NO_PROXY=192.168.99.100
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Verifying Kubernetes components...
* Enabled addons: storage-provisioner, default-storageclass
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2. Play some
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

$ kubectl apply -f configmap.yml -f deployment.yml -f service.yml -f ingress.yml
configmap/my-configmap created
deployment.apps/my-deployment created
service/my-service created
ingress.networking.k8s.io/my-ingress created

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
3. Trying enable ingress
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

$ minikube addons enable ingress
* Verifying ingress addon...

X Exiting due to MK_ENABLE: run callbacks: running callbacks: [waiting for app.kubernetes.io/name=ingress-nginx pods: timed out waiting for the condition]
*
* If the above advice does not help, please let us know:
  - https://github.com/kubernetes/minikube/issues/new/choose

**Comment by zhangb191** [comment]
hi, i got the same problem ,did you fix it? help ~

**Comment by AndreyIvshin** [comment]
I just started to use Ubuntu 20.04 for development

username@devicename:~$ minikube start
😄  minikube v1.17.0 on Ubuntu 20.04
✨  Using the docker driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🔄  Restarting existing docker container for "minikube" ...
🐳  Preparing Kubernetes v1.20.2 on Docker 20.10.2 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
username@devicename:~$ minikube addons enable ingress
🔎  Verifying ingress addon...
🌟  The 'ingress' addon is enabled

**Comment by AndreyIvshin** [comment]
Hello! I had this problem using Docker Toolbox on Windows 10. But after switching on Ubuntu 20.04 it works fine for Docker Engine and Minikube with Docker Driver.

```
username@devicename:~$ minikube start
😄  minikube v1.17.0 on Ubuntu 20.04
✨  Using the docker driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🔄  Restarting existing docker container for "minikube" ...
🐳  Preparing Kubernetes v1.20.2 on Docker 20.10.2 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
username@devicename:~$ minikube addons enable ingress
🔎  Verifying ingress addon...
🌟  The 'ingress' addon is enabled
```

**Comment by priyawadhwa** [comment]
Hey @AndreyIvshin I'm glad you were able to find a solution for this! Looks like it could be a windows specific bug

@zhangb191 could you provide:
-  The OS/driver you are using
-  The  `minikube start` command you are running and its output
- The output of `minikube addons enable ingress --alsologtostderr`
Thank you!

**Comment by yogyogi** [comment]
I am also getting this error in Windows 10 Home. This is how to recreate it.
I am following [this official tutorial](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/) . One I run the below command:
`minikube addons enable ingress`
I get message "After the addon is enabled, please run "minikube tunnel" and your ingress resources would be available at "127.0.0.1". Now when I run
`minikube tunnel`

then everything hangs.

**Comment by spowelljr** [comment]
@yogyogi could you provide the logs @priyawadhwa asked for above?

Detailed logs makes it easier for us to debug the issue, thanks.

**Comment by yogyogi** [comment]
minikube start
`* minikube v1.17.1 on Microsoft Windows 10 Home Single Language 10.0.18363 Build 18363
* Using the docker driver based on existing profile
* Starting control plane node minikube in cluster minikube
* Updating the running docker "minikube" container ...
* Preparing Kubernetes v1.20.2 on Docker 20.10.2 ...
* Verifying Kubernetes components...
* Enabled addons: storage-provisioner, default-storageclass
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by    ###default`

minikube addons enable ingress --alsologtostderr
`I0312 10:21:29.467715    9484 out.go:229] Setting OutFile to fd 92 ...
I0312 10:21:29.470859    9484 out.go:276] TERM=,COLORTERM=, which probably does not support color
I0312 10:21:29.470859    9484 out.go:242] Setting ErrFile to fd 96...
I0312 10:21:29.470859    9484 out.go:276] TERM=,COLORTERM=, which probably does not support color
I0312 10:21:29.489511    9484 addons.go:55] Setting ingress=true in profile "minikube"
I0312 10:21:29.489511    9484 addons.go:131] Setting addon ingress=true in "minikube"
I0312 10:21:29.492814    9484 out.go:119] * After the addon is enabled, please run "minikube tunnel" and your ingress resources would be available at "127.0.0.1"
* After the addon is enabled, please run "minikube tunnel" and your ingress resources would be available at "127.0.0.1"
I0312 10:21:29.493357    9484 host.go:66] Checking if "minikube" exists ...
I0312 10:21:29.520983    9484 cli_runner.go:111] Run: docker container inspect minikube --format={{.State.Status}}
I0312 10:21:29.940140    9484 addons.go:247] installing /etc/kubernetes/addons/ingress-configmap.yaml
I0312 10:21:29.940140    9484 ssh_runner.go:310] scp memory --> /etc/kubernetes/addons/ingress-configmap.yaml (1251 bytes)
I0312 10:21:29.953559    9484 cli_runner.go:111] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube
I0312 10:21:30.380923    9484 sshutil.go:48] new ssh client: &{IP:127.0.0.1 Port:65117 SSHKeyPath:C:\Users\Avita\.minikube\machines\minikube\id_rsa Username:docker}
I0312 10:21:30.473351    9484 addons.go:247] installing /etc/kubernetes/addons/ingress-rbac.yaml
I0312 10:21:30.473351    9484 ssh_runner.go:310] scp memory --> /etc/kubernetes/addons/ingress-rbac.yaml (4828 bytes)
I0312 10:21:30.505030    9484 addons.go:247] installing /etc/kubernetes/addons/ingress-dp.yaml
I0312 10:21:30.505030    9484 ssh_runner.go:310] scp memory --> /etc/kubernetes/addons/ingress-dp.yaml (8533 bytes)
I0312 10:21:30.587432    9484 ssh_runner.go:149] Run: sudo KUBECONFIG=/var/lib/minikube/kubeconfig /var/lib/minikube/binaries/v1.20.2/kubectl apply -f /etc/kubernetes/addons/ingress-configmap.yaml -f /etc/kubernetes/addons/ingress-rbac.yaml -f /etc/kubernetes/addons/ingress-dp.yaml
I0312 10:21:31.167837    9484 addons.go:346] Verifying addon ingress=true in "minikube"
I0312 10:21:31.170832    9484 out.go:119] * Verifying ingress addon...
* Verifying ingress addon...
I0312 10:21:31.194836    9484 kapi.go:75] Waiting for pod with label "app.kubernetes.io/name=ingress-nginx" in ns "kube-system" ...
I0312 10:21:31.226846    9484 kapi.go:86] Found 3 Pods for label selector app.kubernetes.io/name=ingress-nginx
I0312 10:21:31.226846    9484 kapi.go:108] duration metric: took 33.009ms to wait for app.kubernetes.io/name=ingress-nginx ...
I0312 10:21:31.230832    9484 addons.go:97] Writing out "minikube" config to set ingress=true...
I0312 10:21:31.236837    9484 out.go:119] * The 'ingress' addon is enabled
* The 'ingress' addon is enabled`

I have also created this [issue to be looked upon](https://github.com/kubernetes/website/issues/26638#issuecomment-782731795).

**Comment by spowelljr** [comment]
Hi @AndreyIvshin, thanks for reporting your issue!

This is a duplicate of #10544, in order to keep all discussion of this issue centralized I'm going to close this issue.

Please follow along with #10544 and contribute any findings there, thanks for your understanding!
